tests/structured_conf/data/attr_classes.py

```diff
@@ -41,11 +41,26 @@ class UserDict:
     dict: Dict[str, User] = MISSING
 
 
+@attr.s(auto_attribs=True)
+class UserWithDefaultName(User):
+    name: str = "bob"
+
+
```

Comment:
Instead of creating this class, I think you can just merge with `User(name="bob")` in the test.

Reply:
I don't think I can, could you be more specific regarding which test and how to modify it? Currently I don't merge directly with `UserWithDefaultName`, it's used through `MissingUserWithDefaultNameField`.

The specific case I want to test is when we go into this code block:
```python
        if (
            src._is_missing()
            and not dest._is_missing()
            and is_structured_config(src_ref_type)
        )
```
and when `src_ref_type` is such that if I was calling `OmegaConf.structured(src_ref_type)` then I would get an object with at least one non-missing field.

This is a regression test for the bug that was fixed in d686872, where without this change, the test introduced in this commit would fail because `c2.user.name` would be `bob`.

Reply:
You are correct, this is not working here because we are explicitly interesting in the default from the type.